packages/core/src/data-grid/data-grid-types.ts

```diff
@@ -491,6 +491,21 @@ export class CompactSelection {
         return false;
     };
 
+    equals = (other: CompactSelection): boolean => {
+        if (other === this) return true;
+
+        if (other.items.length !== this.items.length) return false;
+
+        for (let i = 0; i < this.items.length; i++) {
+            const left = other.items[i];
+            const right = this.items[i];
+
+            if (left[0] !== right[0] || left[1] || right[1]) return false;
```

Comment:
Pretty sure you meant
```ts
if (left[0] !== right[0] || left[1] !== right[1]) return false;
```

Reply:
Guess I need to figure out why this didn't cause test failures either... it definitely should have

Reply:
eslint should prevent this